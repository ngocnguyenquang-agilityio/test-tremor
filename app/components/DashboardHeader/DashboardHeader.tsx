"use client";

import { useContext, useEffect, useState } from "react";

// Libs
import { usePathname } from "next/navigation";

// Components
import Breadcrumb from "@/components/Breadcrumb/Breadcrumb";

// Icons
import { MdMenu, MdMenuOpen, MdNotifications } from "react-icons/md";
import { HiMiniMoon } from "react-icons/hi2";
import { IoSunny } from "react-icons/io5";

// Constant
import { ROUTES } from "@/constants";

// Contexts
import { ThemeContext } from "../../../src/context/theme";

// Helpers
import { isBrowser } from "@/helpers";

interface DashboardHeaderProps {
  toggleSidebar: () => void;
  isCollapseSidebar: boolean;
}

const DashboardHeader = ({
  isCollapseSidebar,
  toggleSidebar,
}: DashboardHeaderProps): JSX.Element => {
  const { theme, toggleTheme } = useContext(ThemeContext);
  const [isScrolled, setIsScrolled] = useState<boolean>(false);
  const pathname = usePathname();
  const isMobile = isBrowser && window.innerWidth <= 768;
  const isAllProjectPage = pathname === ROUTES.PROJECTS;
  // Check the condition if it is page All Project then display the white color
  const color = (!isScrolled || !isMobile) && isAllProjectPage && "white";

  useEffect(() => {
    const scrollDashboardHeader = () => {
      if (window.scrollY > 0) {
        setIsScrolled(true);
      } else {
        setIsScrolled(false);
      }
    };

    window.addEventListener("scroll", scrollDashboardHeader);

    return () => {
      window.removeEventListener("scroll", scrollDashboardHeader);
    };
  }, []);
  const stickyTransition = "transition-all duration-300 ease-in delay-20";
  const activeStickyScroll =
    isScrolled && (isMobile || !isAllProjectPage)
      ? `sticky top-0 py-2 z-50 bg-lighter dark:bg-dark-gradient-primary/30 dark:bg-dark-gradient-primary/80 box-header-sticky ${stickyTransition} backdrop-saturate-[200%] backdrop-blur-[1.875rem] bg-[rgba(255,255,255,0.8)] min-h-[5rem] rounded-xl top-3 shadow-box-header-sticky`
      : `${stickyTransition}`;

  const activeIconColor =
    isScrolled && (isMobile || !isAllProjectPage)
      ? "text-primary"
      : "text-tremor-content";

  return (
    <div
      className={`${activeStickyScroll} h-32 sm:h-20 mb-3.5 sm:flex items-center justify-between px-2 sm:px-4 py-1 ${
        isAllProjectPage &&
        `absolute top-10 md:top-9 pl-1 pr-2 ${
          activeStickyScroll && isMobile ? "w-full" : "w-[93%]"
        } ${
          isCollapseSidebar
            ? "xl:w-[calc(100%_-_165px)]"
            : "xl:w-[calc(100%_-_330px)]"
        }`
      }`}>
      <div className="flex items-center">
        <div
          className={`block xl:hidden cursor-pointer ${
            isAllProjectPage ? "" : "mr-2"
          }`}
          onClick={toggleSidebar}>
          {isCollapseSidebar ? (
            <MdMenuOpen className="text-2xl" color={color} />
          ) : (
            <MdMenu className="text-2xl" color={color} />
          )}
        </div>
        <Breadcrumb isScrolled={isScrolled} />
      </div>
      <div className="flex items-center justify-between md:items-center mt-4 md:mt-0 md:justify-end pl-6 xl:pl-0">
        <div className="flex flex-wrap item-center gap-y-1">
          <div
            className="sm:block xl:hidden p-2 flex items-center cursor-pointer"
            onClick={toggleSidebar}>
            {isCollapseSidebar ? (
              <MdMenu
                className={`${activeIconColor}  text-2xl`}
                color={color}
              />
            ) : (
              <MdMenuOpen
                className={`${activeIconColor} text-2xl`}
                color={color}
              />
            )}
          </div>
          <div className="relative p-2 flex items-center" onClick={toggleTheme}>
            {theme ? (
              <HiMiniMoon
                className={`${activeIconColor} text-xl cursor-pointer`}
                color={color}
              />
            ) : (
              <IoSunny
                className={`${activeIconColor} text-xl cursor-pointer`}
                color={color}
              />
            )}
          </div>
          <div className="relative p-2 flex items-center">
            <MdNotifications
              className={`${activeIconColor} text-xl cursor-pointer`}
              color={color}
            />
            <div
              className={`absolute top-0 -right-0.5 text-white rounded-full bg-red-500 text-[10px] font-bold py-1 px-2 cursor-pointer leading-none ${
                isAllProjectPage && !isMobile
                  ? "text-white opacity-[0.8]"
                  : "text-inherit"
              }`}>
              9
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default DashboardHeader;
