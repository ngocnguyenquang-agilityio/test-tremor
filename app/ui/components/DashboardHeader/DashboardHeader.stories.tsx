import type { Meta, StoryObj } from "@storybook/react";

// Components
import DashboardHeader from "./DashboardHeader";

const meta = {
  title: "Components/Common/DashboardHeader",
  component: DashboardHeader,
  tags: ["autodocs"],
  argTypes: {
    isCollapseSidebar: {
      description: "Collapse or expand sidebar of dashboard header",
    },
    toggleSidebar: { description: "Toggle sidebar of dashboard header" },
  },
} as Meta<typeof DashboardHeader>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Dashboard: Story = {
  render: () => (
    <div className="bg-body dark:bg-dark-primary rounded-xl shadow-md">
      <DashboardHeader toggleSidebar={() => {}} isCollapseSidebar={false} />
    </div>
  ),
};
