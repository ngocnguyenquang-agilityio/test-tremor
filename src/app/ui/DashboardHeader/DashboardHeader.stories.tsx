import type { Meta, StoryObj } from "@storybook/react";

// Components
import DashboardHeader from "./DashboardHeader";

const meta = {
  title: "Components/DashboardHeader",
  component: DashboardHeader,
  tags: ["autodocs"],
} as Meta<typeof DashboardHeader>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Dashboard: Story = {
  render: () => (
    <DashboardHeader toggleSidebar={() => {}} isCollapseSidebar={false} />
  ),
};
