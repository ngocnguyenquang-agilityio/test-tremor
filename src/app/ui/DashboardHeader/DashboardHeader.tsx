"use client";

import { useEffect, useState } from "react";

// Libs
import { usePathname } from "next/navigation";

// Components
import Breadcrumb from "../../../components/Breadcrumb/Breadcrumb";
import Link from "next/link";
import {
  MdMenu,
  MdMenuOpen,
  MdAccountCircle,
  MdSettings,
  MdNotifications,
} from "react-icons/md";

// Constant
import { ROUTES } from "@/constants";

// Helpers
import { isBrowser } from "@/helpers";

interface DashboardHeaderProps {
  toggleSidebar: () => void;
  isCollapseSidebar: boolean;
}

const DashboardHeader = ({
  isCollapseSidebar,
  toggleSidebar,
}: DashboardHeaderProps): JSX.Element => {
  const [isScrolled, setIsScrolled] = useState<boolean>(false);
  const pathname = usePathname();
  const isMobile = isBrowser && window.innerWidth <= 768;
  const isAllProjectPage = pathname === ROUTES.PROJECTS;
  // Check the condition if it is page All Project then display the white color
  const color = (!isScrolled || !isMobile) && isAllProjectPage && "white";

  useEffect(() => {
    const scrollDashboardHeader = () => {
      if (window.scrollY > 0) {
        setIsScrolled(true);
      } else {
        setIsScrolled(false);
      }
    };

    window.addEventListener("scroll", scrollDashboardHeader);

    return () => {
      window.removeEventListener("scroll", scrollDashboardHeader);
    };
  }, []);
  const stickyTransition = "transition-all duration-300 ease-in delay-20";
  const activeStickyScroll =
    isScrolled && (isMobile || !isAllProjectPage)
      ? `sticky top-0 py-2 z-50 bg-lighter box-header-sticky ${stickyTransition} backdrop-saturate-[200%] backdrop-blur-[1.875rem] bg-[rgba(255,255,255,0.8)] min-h-[5rem] rounded-xl top-3 shadow-box-header-sticky`
      : `${stickyTransition}`;

  const activeIconColor =
    isScrolled && (isMobile || !isAllProjectPage)
      ? "text-primary"
      : "text-tremor-content";

  return (
    <div
      className={`${activeStickyScroll} h-32 md:h-20 mb-3.5 md:flex items-center justify-between px-2 md:px-4 py-1 ${
        isAllProjectPage &&
        `absolute top-10 md:top-9 pl-1 pr-2 ${
          activeStickyScroll && isMobile ? "w-full" : "w-[93%]"
        } ${
          isCollapseSidebar
            ? "xl:w-[calc(100%_-_165px)]"
            : "xl:w-[calc(100%_-_330px)]"
        }`
      }`}>
      <div className="flex items-center">
        <Breadcrumb isScrolled={isScrolled} />
        <div
          className="hidden xl:block cursor-pointer mx-16"
          onClick={toggleSidebar}>
          {isCollapseSidebar ? (
            <MdMenuOpen
              className="text-tremor-content text-2xl"
              color={color}
            />
          ) : (
            <MdMenu className="text-tremor-content text-2xl" color={color} />
          )}
        </div>
      </div>
      <div className="flex items-center justify-between md:items-center mt-4 md:mt-0 md:justify-end">
        <div className="flex flex-wrap item-center gap-y-1">
          <Link
            href={ROUTES.SIGN_IN}
            aria-label="Link to homepage"
            className="p-2 flex items-center">
            <MdAccountCircle
              className={`${activeIconColor} text-xl`}
              color={color}
            />
          </Link>
          <div
            className="sm:block xl:hidden p-2 flex items-center cursor-pointer"
            onClick={toggleSidebar}>
            {isCollapseSidebar ? (
              <MdMenu
                className={`${activeIconColor}  text-2xl`}
                color={color}
              />
            ) : (
              <MdMenuOpen
                className={`${activeIconColor} text-2xl`}
                color={color}
              />
            )}
          </div>
          <div className="relative p-2 flex items-center">
            <MdSettings
              className={`${activeIconColor} text-xl cursor-pointer`}
              color={color}
            />
          </div>
          <div className="relative p-2 flex items-center">
            <MdNotifications
              className={`${activeIconColor} text-xl cursor-pointer`}
              color={color}
            />
            <div
              className={`absolute top-0 -right-0.5 text-white rounded-full bg-red-500 text-[10px] font-bold py-1 px-2 cursor-pointer leading-none ${
                isAllProjectPage && !isMobile
                  ? "text-white opacity-[0.8]"
                  : "text-inherit"
              }`}>
              9
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default DashboardHeader;
