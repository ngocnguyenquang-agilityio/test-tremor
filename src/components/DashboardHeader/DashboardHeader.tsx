"use client";

// Libs
import { useContext } from "react";
import { usePathname } from "next/navigation";

// Components
import BreadCrumb from "../BreadCrumb/BreadCrumb";
import { TextInput } from "@tremor/react";
import Link from "next/link";
import {
  MdMenu,
  MdMenuOpen,
  MdAccountCircle,
  MdSettings,
  MdNotifications,
} from "react-icons/md";

// Constant
import { ROUTES } from "@/constants";

// Context
import { SidebarContext } from "@/contexts";

const DashboardHeader = (): JSX.Element => {
  const { isOpen, toggleSideBar } = useContext(SidebarContext);

  const pathname = usePathname();
  const isProjectPage = pathname === ROUTES.PROJECTS;

  return (
    <div
      className={`md:flex items-center justify-between px-4 py-1 ${
        isProjectPage
          ? `absolute top-8 w-[93%] ${
              isOpen ? "xl:w-[calc(100%_-_165px)]" : "xl:w-[calc(100%_-_330px)]"
            }`
          : ""
      }`}>
      <div className="flex items-center">
        <BreadCrumb />
        <div
          className="hidden xl:block cursor-pointer mx-16"
          onClick={toggleSideBar}>
          {isOpen ? (
            <MdMenuOpen
              className="text-tremor-content text-2xl"
              color={`${isProjectPage ? "white" : "inherit"}`}
            />
          ) : (
            <MdMenu
              className="text-tremor-content text-2xl"
              color={`${isProjectPage ? "white" : "inherit"}`}
            />
          )}
        </div>
      </div>
      <div className="flex items-center mt-4 md:mt-0 mr-3 justify-end">
        <div className="mr-4">
          <TextInput
            id="search"
            placeholder="Search here"
            type="text"
            autoFocus
            className="py-0.5 w-full bg-transparent hover:bg-transparent focus:bg-transparent min-w-[100px]"
          />
        </div>
        <div className="flex">
          <Link href={ROUTES.SIGN_IN} aria-label="Link to homepage">
            <MdAccountCircle
              className="text-tremor-content text-xl"
              color={`${isProjectPage ? "white" : "inherit"}`}
            />
          </Link>
          <div
            className="sm:block xl:hidden cursor-pointer ml-4"
            onClick={toggleSideBar}>
            {isOpen ? (
              <MdMenuOpen
                className="text-tremor-content text-2xl"
                color={`${isProjectPage ? "white" : "inherit"}`}
              />
            ) : (
              <MdMenu
                className="text-tremor-content text-2xl"
                color={`${isProjectPage ? "white" : "inherit"}`}
              />
            )}
          </div>
          <div className="relative">
            <MdSettings
              className="text-tremor-content text-xl mx-4 cursor-pointer"
              color={`${isProjectPage ? "white" : "inherit"}`}
            />
          </div>
          <div className="relative max-w-[10px]">
            <MdNotifications
              className="text-tremor-content text-xl cursor-pointer"
              color={`${isProjectPage ? "white" : "inherit"}`}
            />
            <div
              className={`absolute -top-3 -right-3 text-white rounded-full bg-red-500 text-[10px] font-bold py-1 px-2 cursor-pointer ${
                isProjectPage ? "text-white opacity-[0.8]" : "text-inherit"
              }`}>
              9
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default DashboardHeader;
