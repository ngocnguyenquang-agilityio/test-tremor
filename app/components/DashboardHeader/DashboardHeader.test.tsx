import { render } from "@testing-library/react";

// Components
import DashboardHeader from "./DashboardHeader";

describe("DashboardHeader component", () => {
  const dashboardHeader = () => {
    return render(
      <DashboardHeader toggleSidebar={() => {}} isCollapseSidebar={false} />,
    );
  };

  it("Should render DashboardHeader snapshot correctly", () => {
    expect(dashboardHeader()).toMatchSnapshot();
  });
});
