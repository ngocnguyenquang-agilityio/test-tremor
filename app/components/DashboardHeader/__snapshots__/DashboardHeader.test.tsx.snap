// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardHeader component Should render DashboardHeader snapshot correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="transition-all duration-300 ease-in delay-20 h-32 sm:h-20 mb-3.5 sm:flex items-center justify-between px-2 sm:px-4 py-1 false"
      >
        <div
          class="flex items-center"
        >
          <div
            class="block xl:hidden cursor-pointer mr-2"
          >
            <svg
              class="text-2xl"
              color="#344767"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(52, 71, 103);"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </div>
          <nav
            class=""
          >
            <ol
              class="flex flex-wrap items-center text-gray-400"
            >
              <li>
                <a
                  class="flex"
                  href="/dashboards/analytics"
                >
                  <svg
                    class="bg-inherit border-0 py-0 hover:bg-transparent text-primary dark:text-dark-primary tracking-[0.01071em] leading-[0] text-inherit opacity-50"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                    <path
                      d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                    />
                  </svg>
                </a>
              </li>
              <li
                aria-hidden="true"
                class="text-sm mx-2 text-[#6c757d]"
              >
                /
              </li>
              <li
                class="flex items-center text-inherit"
              >
                <div
                  class="bc-link"
                >
                  <a
                    class="text-sm capitalize tracking-[0.02857em] leading-[0] text-primary dark:text-dark-primary opacity-50"
                    href="/dashboards/analytics"
                  >
                    dashboards
                  </a>
                </div>
                <div
                  aria-hidden="true"
                  class="text-sm mx-2 text-[#6c757d]"
                >
                  /
                </div>
              </li>
              <li
                class="text-sm text-tremor-content-title dark:text-dark-tremor-content-title capitalize tracking-[0.02857em] leading-[0] text-inherit"
              >
                analytics
              </li>
            </ol>
            <p
              class="text-tremor-title dark:text-dark-tremor-content-title font-bold capitalize tracking-[0.0075em] text-primary"
            >
              analytics
            </p>
          </nav>
        </div>
        <div
          class="flex items-center justify-between md:items-center mt-4 md:mt-0 md:justify-end pl-6 xl:pl-0"
        >
          <div
            class="flex flex-wrap item-center gap-y-1"
          >
            <div
              class="relative p-2 flex items-center"
            >
              <svg
                class="text-tremor-content text-xl cursor-pointer"
                color="#344767"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="color: rgb(52, 71, 103);"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 118a22 22 0 01-22-22V48a22 22 0 0144 0v48a22 22 0 01-22 22zm0 368a22 22 0 01-22-22v-48a22 22 0 0144 0v48a22 22 0 01-22 22zm113.14-321.14a22 22 0 01-15.56-37.55l33.94-33.94a22 22 0 0131.11 31.11l-33.94 33.94a21.93 21.93 0 01-15.55 6.44zM108.92 425.08a22 22 0 01-15.55-37.56l33.94-33.94a22 22 0 1131.11 31.11l-33.94 33.94a21.94 21.94 0 01-15.56 6.45zM464 278h-48a22 22 0 010-44h48a22 22 0 010 44zm-368 0H48a22 22 0 010-44h48a22 22 0 010 44zm307.08 147.08a21.94 21.94 0 01-15.56-6.45l-33.94-33.94a22 22 0 0131.11-31.11l33.94 33.94a22 22 0 01-15.55 37.56zM142.86 164.86a21.89 21.89 0 01-15.55-6.44l-33.94-33.94a22 22 0 0131.11-31.11l33.94 33.94a22 22 0 01-15.56 37.55zM256 358a102 102 0 11102-102 102.12 102.12 0 01-102 102z"
                />
              </svg>
            </div>
            <div
              class="relative p-2 flex items-center"
            >
              <svg
                class="text-tremor-content text-xl cursor-pointer"
                color="#344767"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="color: rgb(52, 71, 103);"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 22c1.1 0 2-.9 2-2h-4a2 2 0 002 2zm6-6v-5c0-3.07-1.64-5.64-4.5-6.32V4c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5v.68C7.63 5.36 6 7.92 6 11v5l-2 2v1h16v-1l-2-2z"
                />
              </svg>
              <div
                class="absolute top-0 -right-0.5 text-white rounded-full bg-red-500 text-[10px] font-bold py-1 px-2 cursor-pointer leading-none text-inherit"
              >
                9
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="transition-all duration-300 ease-in delay-20 h-32 sm:h-20 mb-3.5 sm:flex items-center justify-between px-2 sm:px-4 py-1 false"
    >
      <div
        class="flex items-center"
      >
        <div
          class="block xl:hidden cursor-pointer mr-2"
        >
          <svg
            class="text-2xl"
            color="#344767"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            style="color: rgb(52, 71, 103);"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </div>
        <nav
          class=""
        >
          <ol
            class="flex flex-wrap items-center text-gray-400"
          >
            <li>
              <a
                class="flex"
                href="/dashboards/analytics"
              >
                <svg
                  class="bg-inherit border-0 py-0 hover:bg-transparent text-primary dark:text-dark-primary tracking-[0.01071em] leading-[0] text-inherit opacity-50"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                  />
                </svg>
              </a>
            </li>
            <li
              aria-hidden="true"
              class="text-sm mx-2 text-[#6c757d]"
            >
              /
            </li>
            <li
              class="flex items-center text-inherit"
            >
              <div
                class="bc-link"
              >
                <a
                  class="text-sm capitalize tracking-[0.02857em] leading-[0] text-primary dark:text-dark-primary opacity-50"
                  href="/dashboards/analytics"
                >
                  dashboards
                </a>
              </div>
              <div
                aria-hidden="true"
                class="text-sm mx-2 text-[#6c757d]"
              >
                /
              </div>
            </li>
            <li
              class="text-sm text-tremor-content-title dark:text-dark-tremor-content-title capitalize tracking-[0.02857em] leading-[0] text-inherit"
            >
              analytics
            </li>
          </ol>
          <p
            class="text-tremor-title dark:text-dark-tremor-content-title font-bold capitalize tracking-[0.0075em] text-primary"
          >
            analytics
          </p>
        </nav>
      </div>
      <div
        class="flex items-center justify-between md:items-center mt-4 md:mt-0 md:justify-end pl-6 xl:pl-0"
      >
        <div
          class="flex flex-wrap item-center gap-y-1"
        >
          <div
            class="relative p-2 flex items-center"
          >
            <svg
              class="text-tremor-content text-xl cursor-pointer"
              color="#344767"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(52, 71, 103);"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 118a22 22 0 01-22-22V48a22 22 0 0144 0v48a22 22 0 01-22 22zm0 368a22 22 0 01-22-22v-48a22 22 0 0144 0v48a22 22 0 01-22 22zm113.14-321.14a22 22 0 01-15.56-37.55l33.94-33.94a22 22 0 0131.11 31.11l-33.94 33.94a21.93 21.93 0 01-15.55 6.44zM108.92 425.08a22 22 0 01-15.55-37.56l33.94-33.94a22 22 0 1131.11 31.11l-33.94 33.94a21.94 21.94 0 01-15.56 6.45zM464 278h-48a22 22 0 010-44h48a22 22 0 010 44zm-368 0H48a22 22 0 010-44h48a22 22 0 010 44zm307.08 147.08a21.94 21.94 0 01-15.56-6.45l-33.94-33.94a22 22 0 0131.11-31.11l33.94 33.94a22 22 0 01-15.55 37.56zM142.86 164.86a21.89 21.89 0 01-15.55-6.44l-33.94-33.94a22 22 0 0131.11-31.11l33.94 33.94a22 22 0 01-15.56 37.55zM256 358a102 102 0 11102-102 102.12 102.12 0 01-102 102z"
              />
            </svg>
          </div>
          <div
            class="relative p-2 flex items-center"
          >
            <svg
              class="text-tremor-content text-xl cursor-pointer"
              color="#344767"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(52, 71, 103);"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 22c1.1 0 2-.9 2-2h-4a2 2 0 002 2zm6-6v-5c0-3.07-1.64-5.64-4.5-6.32V4c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5v.68C7.63 5.36 6 7.92 6 11v5l-2 2v1h16v-1l-2-2z"
              />
            </svg>
            <div
              class="absolute top-0 -right-0.5 text-white rounded-full bg-red-500 text-[10px] font-bold py-1 px-2 cursor-pointer leading-none text-inherit"
            >
              9
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
